plugins/interactive/interactive.go

```diff
@@ -0,0 +1,287 @@
+package interactive
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	schematypes "github.com/taskcluster/go-schematypes"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/plugins"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/ioext"
+)
+
+// defaultArtifactPrefix is the default artifact prefix used if nothing is
+// configured or given in the task definition
+const defaultArtifactPrefix = "private/interactive/"
+
+// defaultShellToolURL is the default URL for the tool that can connect to the
+// shell socket and display an interactive shell session.
+const defaultShellToolURL = "https://tools.taskcluster.net/shell/"
+
+// defaultShellToolURL is the default URL for the tool that can list displays
+// and connect to the display socket with interactive noVNC session.
+const defaultDisplayToolURL = "https://tools.taskcluster.net/display/"
+
+type provider struct {
+	plugins.PluginProviderBase
+}
+
+func (provider) ConfigSchema() schematypes.Schema {
+	return configSchema
+}
+func (provider) NewPlugin(options plugins.PluginOptions) (plugins.Plugin, error) {
+	var c config
+	if schematypes.MustMap(configSchema, options.Config, &c) != nil {
```

Comment:
No need, if input doesn't validate it's a contract violation.

The worker is responsible for validating against the schema before passing it to the plugin or engines. So engines/plugins should treat violation as a contract violation.